### DISCUSSION TITLE
Early static API design decisions for JsonSerializer in System.Text.Json

### DISCUSSION THREAD

**Discussion by nikouu** [dotnet/runtime#107496] [Q&A] [answered]
I've been looking toward `System.Text.Json` for utility class design inspiration and I'm keen to understand what went into the early design for the static APIs (i.e. `JsonSerializer.*`). 

Finding early discussions around the static APIs is a bit difficult as this work began in the corefx repo outside of the usual runtime API discussions. Some relevant references I could find include:

- [SerializerProgrammingModel.md](https://github.com/dotnet/corefx/blob/master/src/System.Text.Json/docs/SerializerProgrammingModel.md)
- [`JsonSerializer` initial commit](https://github.com/dotnet/corefx/pull/35609/)
- [Announcement to move off Json.NET](https://github.com/dotnet/runtime/issues/27761)
- [Various runtime `System.Text.Json` reviews in 2018](https://github.com/dotnet/apireviews/tree/main/2018)

[Looking way back in Json.NET too](https://github.com/JamesNK/Newtonsoft.Json/blame/5bb0c17eecd9f0bef60df0142d66595227685016/Src/Newtonsoft.Json/JavaScriptConvert.cs#L437) didn't seem to yield answers either.

I did find unverifiable answers, often overlapping with general reasoning around `static`:

1. `static` is more performant 
2. It's a utility class so it should be `static`
3. `System.Text.Json` is `static` to be a familiar API to Json.NET

To sum up: I'd like to know the initial `static`-making decision behind `JsonSerializer.Serialize()`, `JsonSerializer.Deserialize()`, and friends - preferably with references (or from the developers themselves).

**Comment by huoyaoyuan** [comment]
@eiriktsarpalis 

I believe it's more about state. `static` means no states are preserved. However, `JsonSerializer` does need to keep some states to materialize the options, so the `JsonSerializationOptions` needs to be manually preserved. This has caused several problems and #74492 is exploring stateful `JsonSerializer` instances.

**Reply by eiriktsarpalis** [reply]
I'd have to defer to @terrajobst and @bartonjs on the original reasons why `JsonSerializer` exposes static APIs, but I would guess it has something to do with being able to quickly serialize objects into JSON without a lot of ceremony.

**Reply by terrajobst** [reply, marked as answer]
@eiriktsarpalis is correct: the goal was to minimize ceremony to have simple one-liners. I believe we modeled it off of `Newtonsoft.Json`'s `JsonConvert` API.

**Reply by nikouu** [reply]
Fantastic, thank you everyone!
